algo/src/main/java/com/hartwig/serve/extraction/gene/GeneLevelExtractor.java

```diff
@@ -54,14 +54,22 @@ public GeneLevelExtractor(@NotNull final GeneChecker exomeGeneChecker, @NotNull
 
     @Nullable
     public GeneAnnotation extract(@NotNull String gene, @NotNull EventType type, @NotNull String event) {
-        if (type == EventType.WILD_TYPE && exomeGeneChecker.isValidGene(gene)) {
-            return extractWildTypeEvent(gene, type);
-        } else if (type == EventType.GENE_LEVEL && exomeGeneChecker.isValidGene(gene)) {
-            return extractGeneLevelEvent(gene, event);
-        } else if (type == EventType.PROMISCUOUS_FUSION && fusionGeneChecker.isValidGene(gene)) {
-            return extractPromiscuousFusion(gene);
+        if (exomeGeneChecker.isValidGene(gene) || fusionGeneChecker.isValidGene(gene)) {
+            switch (type) {
+                case WILD_TYPE:
+                    return extractWildTypeEvent(gene, type);
+                case GENE_LEVEL:
+                    return extractGeneLevelEvent(gene, event);
+                case PROMISCUOUS_FUSION:
+                    return extractPromiscuousFusion(gene);
+                case ABSENCE_OF_PROTEIN:
+                case PRESENCE_OF_PROTEIN:
+                    return extractProteinEvent(gene, type);
+                default:
+                    LOGGER.warn("Unrecognized event type : '{}'", type);
+                    return null;
+            }
```

Comment:
With this change, you run the risk of sending a valid fusion gene but not exome gene into e.g. `GENE_LEVEL` branch. This shouldn't be possible!

Reply:
I have revert it back